src/Etg.Yams.Powershell/InstallApplicationsCmdlet.cs

```diff
@@ -125,8 +125,15 @@ protected override void ProcessRecord()
                         {
                             deploymentConfig = deploymentConfig.RemoveApplication(appId);
                         }
-                        var newAppIdentity = new AppIdentity(appId, version);
-                        deploymentConfig = deploymentConfig.AddApplication(newAppIdentity, clusterId);
+
+                        if (RemoveOldVersions || 
```

Comment:
What about?
```csharp
if (RemoveOldVersions && deploymentConfig.HasApplication(appId))
{		                        
	deploymentConfig = deploymentConfig.RemoveApplication(appId);
}
else if (deploymentConfig.ListVersions(appId).Contains(version) == false)
{
	var newAppIdentity = new AppIdentity(appId, version);
	deploymentConfig = deploymentConfig.AddApplication(newAppIdentity, clusterId);
}
```

Reply:
Also you can call `deploymentConfig.HasApplication(appIdentity)` instead of using `ListVersions`.

Reply:
@nehmebilal Sounds good, I'll change to `HasApplication`.

I didn't want to make it `else if` because then if it removes the old versions, it will not add the new version.


Case 1 - Install 1.0 on top of 0.9 with 'RemoveOldApplications' = true
What happens: 0.9 removed, 1.0 installed

Case 2 - Install 1.0 on top of 1.0 with 'RemoveOldApplications' = true
Fails

Case 3 - Install 1.0 on top of 0.9 with 'RemoveOldApplications' = false
What happens: 1.0 should be  installed, 0.9 should be retained.

Case 4 - Install 1.0 on top of 1.0 with 'RemoveOldApplications' = false
What happens: Do nothing


Currently Case 2 is failing.

My proposed change: 1.0 removed, 1.0 installed
Your change: 1.0 is removed (but not installed)

Reply:
Ok makes sense!

Not a big deal but it's probably better remove the second `RemoveOldVersions` from the second if since we're not listing anymore (it's pretty fast to check and better to keep it simple). What do you think?

Reply:
Sure, if it's a quick enough check its maybe a little more clear.